extension/solarwindsapmsettingsextension/extension.go

```diff
@@ -25,13 +32,63 @@ func newSolarwindsApmSettingsExtension(extensionCfg *Config, logger *zap.Logger)
 	return settingsExtension, nil
 }
 
-func (extension *solarwindsapmSettingsExtension) Start(_ context.Context, _ component.Host) error {
+func (extension *solarwindsapmSettingsExtension) Start(ctx context.Context, _ component.Host) error {
 	extension.logger.Debug("Starting up solarwinds apm settings extension")
-	_, extension.cancel = context.WithCancel(context.Background())
+	ctx, extension.cancel = context.WithCancel(ctx)
+	configOk := validateSolarwindsApmSettingsExtensionConfiguration(extension.config, extension.logger)
+	if configOk {
+		extension.conn, _ = grpc.Dial(extension.config.Endpoint, grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{})))
```

Comment:
you are ignoring the error returned by this function.

Reply:
I fixed it by returning the error.